src/grading-settings/assignment-section/AssignmentSection.test.jsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { render, waitFor, fireEvent } from '@testing-library/react';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+
+import AssignmentSection from '.';
+import messages from './messages';
+
+const testObj = {};
+
+const defaultAssignments = {
+  type: 'Test type',
+  minCount: 1,
+  dropCount: 1,
+  shortLabel: 'TT',
+  weight: 100,
+  id: 0,
+};
+
+const setGradingData = (fn) => {
+  testObj.graders = fn({}).graders;
+};
+
+const RootWrapper = (props = {}) => (
+  <IntlProvider locale="en">
+    <AssignmentSection
+      handleRemoveAssignment={jest.fn()}
+      setShowSavePrompt={jest.fn()}
+      graders={[defaultAssignments]}
+      setGradingData={jest.fn()}
+      courseAssignmentLists={defaultAssignments}
+      setShowSuccessAlert={jest.fn()}
+      {...props}
+    />
+  </IntlProvider>
+);
+
+describe('<AssignmentSection />', () => {
+  it('checking the correct display of titles, labels, descriptions', async () => {
+    const { getByText } = render(<RootWrapper />);
+    await waitFor(() => {
+      expect(getByText(messages.assignmentTypeNameTitle.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.assignmentTypeNameDescription.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.abbreviationTitle.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.abbreviationDescription.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.weightOfTotalGradeTitle.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.weightOfTotalGradeDescription.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.totalNumberTitle.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.totalNumberDescription.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.numberOfDroppableTitle.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.numberOfDroppableDescription.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.assignmentAlertWarningDescription.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.assignmentDeleteButton.defaultMessage)).toBeInTheDocument();
+      expect(getByText('Warning: The number of Test type assignments defined here does not match the current number of Test type assignments in the course:')).toBeInTheDocument();
```

Comment:
should this text be translatable?

Reply:
We have translations for this text
<img width="965" alt="image" src="https://github.com/raccoongang/frontend-app-course-authoring/assets/93188219/02b17470-da2a-4cfd-b13a-dea922460d4c">

Reply:
We decided to use for such cases a wrapper of the format: `{intl.formatMessage(messages.assignmentAlertWarningUsageTitle, { type: gradeField.type })}`

Corrected, please take a look